crates/terminal_view/src/terminal_view.rs

```diff
@@ -576,15 +578,25 @@ fn possible_open_targets(
         } else {
             Vec::new()
         }
-    } else if let Some(workspace) = workspace.upgrade() {
-        workspace.update(cx, |workspace, cx| {
-            workspace
-                .worktrees(cx)
-                .map(|worktree| worktree.read(cx).abs_path().join(&maybe_path))
-                .collect()
-        })
     } else {
-        Vec::new()
+        // First check cwd and then workspace
+        let mut potential_cwd_and_workspace_paths = Vec::new();
+
+        if let Some(cwd) = cwd {
+            potential_cwd_and_workspace_paths.push(Path::join(cwd, maybe_path.clone()));
+        }
+
+        if let Some(workspace) = workspace.upgrade() {
+            let workspace_path = workspace.update(cx, |workspace, cx| {
+                workspace
+                    .worktrees(cx)
+                    .map(|worktree| worktree.read(cx).abs_path().join(&maybe_path))
+                    .collect()
+            });
+            potential_cwd_and_workspace_paths.push(workspace_path);
```

Comment:
This is wrong: now the code gets all worktrees, gets their abs path, joins with the `maybe_path` and takes all those joins and joins them again into one path.

Instead, we're supposed to keep the old code that forms a `Vec<PathBuf>` and add one more there, if the `cwd` exists.